crates/qasm3/src/build.rs

```diff
@@ -377,37 +372,45 @@ pub fn convert_asg(
         pygates: gate_constructors,
         module,
     };
+
+    let _ = state.map_gate_ids(py, ast_symbols);
+
```

Comment:
This is explicitly ignoring a `Result`, with the one bit of syntax that gets around the clippy lint for not "using" a `Result` return value.  I think what's intended is `state.map_gate_ids(py, ast_symbols)?;` with no `let`.   It'd be good to add a test that the exceptions are raised in Python space - we didn't do it at the initial merge because of a rush to land the PR, but we ought to back fill.

Reply:
I thought clippy complains about throwing away *any* return value. I agree that `state.map_gate_ids(py, ast_symbols)?` is much better.

Reply:
Partly fixed in 258499c
But I did not yet write a new test.

Reply:
`Result` as a type is marked `#[must_use]`, so any expression that evaluates to a `Result` needs to be "used" in some form.  Most types don't have that annotation, though some functions returning normally safe types might have it, so you may have seen it there.